docs/source/developer/architecture/5_building_block_view.rst

```diff
@@ -7,10 +7,64 @@
 5. Building Block View
 ======================
 
+The layers of fotoobo
+---------------------
+
+There are three designated layers in **fotoobo**. These are the interface layer, the business logic
+layer and the infrastructure layer.
+
+.. image:: diagrams/fotoobo_layers.drawio.svg
+  :width: 100%
+  :alt: The layers of fotoobo visualized
+
+In this diagram you see the designated paths to use when accessing a Fortinet device. Although these 
+are the recommended ways to use it, you may also access the layers in other ways. Just keep in mind 
+there always should be a rational and understandable approach.
+
+
+The Interface Layer
+^^^^^^^^^^^^^^^^^^^
+
+This is the main entry point for interaction with **fotoobo** when you installed it as an
+application on your system. It acts as a frontend for users and automated tools. At the Moment only
+the CLI part is implemented. In future version an API may be available if there is a need for it.
```

Comment:
I would be more precise here: "a REST API", because we kinda already have an API, in Python for programmers (the "Business Logic Layer"), but it still is an API too ;-)

Reply:
Yes, way better than just "API"